testutils/clustermanager/perf-tests/pkg/cluster.go

```diff
@@ -44,8 +46,9 @@ type gkeClient struct {
 }
 
 // NewClient will create a new gkeClient.
-func NewClient() (*gkeClient, error) {
-	operations, err := gke.NewSDKClient()
+func NewClient(environment string) (*gkeClient, error) {
+	endpointOption := option.WithEndpoint(gke.ServiceEndpoint(environment))
```

Comment:
You should validate the return of ServiceEndpoint and return a meaningful error.

Reply:
I changed the logic a bit:
If the provided gke environment is empty or invalid, default the endpoint to be prod. So the return of ServiceEndpoint will be always valid, unless caller provides some custom endpoint like `https://invalid.endpoint`.

Does it make sense to you?

Reply:
I don't think an invalid endpoint (which includes the empty string) should be silently translated to prod, it's dangerous and counter-intuitive. Anyone calling NewClient() should be sure about the environment. If you want to keep backwards compatibility, rename this to NewClientForEnvironment() and make NewClient() call this method passing prod. But since this method is not widely used yet, I don't think you need to do such thing.

Reply:
Make sense.
Added error handling.